src/components/radio-group/daikin-radio-group.ts

```diff
@@ -0,0 +1,203 @@
+  private readonly _radios!: Array<DaikinRadio>;
+
+  /**
+   * Specify the radio group orientation
+   */
+  @property({ type: String })
+  orientation: RadioGroupProps["orientation"] = "horizontal";
+
+  /**
+   * The form name.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * `value` of the currently selected radio.
+   * see {@link DaikinRadio.value}
+   */
+  @property({ type: String, reflect: true })
+  value = "";
+
+  private _updateRadios() {
+    const radios = this._radios;
+    const selectedRadio = radios.find((radio) => radio.value === this.value);
+    for (const [, daikinRadio] of this._radios.entries()) {
+      if (this.name) {
+        daikinRadio.name = this.name;
+      }
+      const isSelected = daikinRadio === selectedRadio;
+      daikinRadio.checked = isSelected;
+      daikinRadio.skipTab = !isSelected;
+      daikinRadio.internals.setFormValue(isSelected ? daikinRadio.value : null);
+    }
+  }
+
+  private readonly _handleRadioChange = (
+    event: CustomEvent<{ value: string }>
+  ) => {
+    this.value = (event.target as HTMLInputElement).value;
+  };
+
+  private _handleSlotChange(): void {
+    this._updateRadios();
+  }
+
+  /**
+   * Handles keyboard interactions in the radio list.
+   * https://www.w3.org/WAI/ARIA/apg/patterns/radio/
+   */
+  private _handleKeyDown(event: KeyboardEvent): void {
+    const moveOffset = {
+      ArrowRight: 1,
+      ArrowDown: 1,
+      ArrowLeft: -1,
+      ArrowUp: -1,
+    }[event.key];
+    if (!moveOffset) {
+      return;
+    }
+
+    // Retrieve all radios
+    const radios = this._radios;
+
+    // Check if there is at least one radio available
+    if (!radios.some((radio) => !radio.disabled)) {
+      // No radios available!
+      if (import.meta.env.DEV) {
+        console.warn(
+          `[daikin-radio-group] No radios that can be activated! This may cause unexpected behavior.`
+        );
+      }
+      return;
+    }
+
+    // Get focused radio if any
+    const activeElement = document.activeElement;
+    const focusedRadioIndex = activeElement
+      ? radios.findIndex((radio) => radio.contains(activeElement))
+      : -1;
+
+    // If there is no radio focused, focus on the active (current) radio
+    if (focusedRadioIndex < 0) {
+      const activeRadio = radios.find(
+        (radio) => !radio.disabled && radio.checked
+      );
+      activeRadio?.shadowRoot?.querySelector("input")?.focus();
+      event.preventDefault();
+      return;
+    }
+
+    // If there is a radio focused, move focus forward or backward
+    for (let i = 1; i <= radios.length; i++) {
+      const index =
+        (focusedRadioIndex + moveOffset * i + radios.length * i) %
+        radios.length;
+      const nextRadio = radios[index];
+      if (nextRadio.disabled) {
+        continue;
+      }
+      const nextRadioInput = nextRadio.shadowRoot?.querySelector("input");
+      nextRadioInput?.focus();
```

Comment:
Implement focus delegation to `daikin-radio` component.


```suggestion
      nextRadio.focus();
```